src/main.rs

```diff
@@ -72,13 +79,13 @@ fn to_expr(v: Value) -> Result<Box<Expr>, &'static String> {
                     "=" => Expr::EqualTo,
                     _ => panic!(), // Unimplemented operand
                 };
-                expr(to_expr(x.to_owned())?, to_expr(y.to_owned())?)
+                expr(to_expr(x.take())?, to_expr(y.take())?)
             }
             // Bad way of checking if it's a Field type
             // TODO, handle the error
             [String(s), Number(i)] => Expr::Field(i.as_i64().unwrap() as i32),
             // Bad way of checking if it's a macro
-            [String(s), String(n)] => Expr::Macro(n.to_owned()),
+            [String(s), String(n)] => Expr::Macro(n.to_string()),
```

Comment:
I think this is wrong, because I imagine if you call to_string on a string reference it copies the original string instead of consuming it like I'm trying to do here.

Reply:
> I think this is wrong, because I imagine if you call to_string on a string reference it copies the original string instead of consuming it like I'm trying to do here.

Yes. It's pointless, as `s` and `n` are already `String` I guess, not `&str`? Not sure though. The compiler should tell you that.

Reply:
s & n are actually `&std::string::String` @snawaz